RUFAS/input_manager.py

```diff
@@ -756,259 +731,337 @@ def _add_default_values_to_array_inputs(
+        complex_type_to_validator_map: Dict[
+            str, Callable[[List[int | str], Dict[str, Any], Dict[str, Any], bool, str, "ElementsCounter", bool], bool]
+        ] = {
+            "array": self._array_type_validator,
+            "object": self._object_type_validator,
+        }
+
+        if data_type not in primitive_data_type_to_validator_map and data_type not in complex_type_to_validator_map:
+            raise ValueError(
+                f"The metadata type of the element '{self._convert_variable_path_to_str(variable_path)}' "
+                f"is not valid. Supported types are: {primitive_data_type_to_validator_map.keys()}, "
+                f"{complex_type_to_validator_map.keys()}"
+            )
+
+        elif data_type in primitive_data_type_to_validator_map:
+            variable_path_str = self._convert_variable_path_to_str(variable_path)
+            input_data_value = self._extract_value_by_key_list(input_data, variable_path)
+            is_valid = primitive_data_type_to_validator_map[data_type](
+                variable_properties,
+                variable_path_str,
+                input_data_value,
+                properties_blob_key,
+            )
+            if is_valid:
+                elements_counter.increment(ElementState.VALID)
+                return True
+            is_fixed = self._fix_data(variable_properties, variable_path, input_data, properties_blob_key)
+            if is_fixed:
+                elements_counter.increment(ElementState.FIXED)
+                return True
+            elements_counter.increment(ElementState.INVALID)
+            return False
+
+        else:
+            return complex_type_to_validator_map[data_type](
+                variable_path,
+                variable_properties,
+                input_data,
+                eager_termination,
+                properties_blob_key,
+                elements_counter,
+                called_during_initialization,
+            )
+
+    def _dict_type_validator(
         self,
-        element_hierarchy: List[str],
+        first_level_key: str,
         properties_blob_key: str,
         input_data: Dict[str, Any],
         eager_termination: bool,
-        element_counter_and_validity: Dict[str, int | bool],
+        elements_counter: "ElementsCounter",
         called_during_initialization: bool = False,
-    ) -> Dict[str, int | bool]:
+    ) -> bool:
         """
         Receives data loaded from json input file, recursively finds and then validates nested elements,
         attempts to fix any invalid elements, and tracks the number of valid, invalid, fixed,
         and total elements from the input file are checked.
 
         Parameters
         ----------
-        element_hierarchy : List[str]
-            A list of strings representing the path to the data being validated.
+        first_level_key : str
+            The name of the json data element just under the level of properties_blob_key.
```

Comment:
```suggestion
            The name of the data element just under the level of properties_blob_key.
```
Two things here:
1. It would be incorrect to say that this is came from a JSON, it isn't known in the context of this function where it came from.
2. Could please reword this description or give some examples? I am not sure what you mean by this.